src/asset-relocator.js

```diff
@@ -804,6 +838,11 @@ module.exports = async function (content, map) {
         // if we have a direct pure static function,
         // and that function has a [TRIGGER] symbol -> trigger asset emission from it
         if (calleeValue && typeof calleeValue.value === 'function' && calleeValue.value[TRIGGER]) {
+          // Skip static analysis for write operations -> to replace for example __dirname
+          // It's not possible analyze not existing files
+          if(options.writeMode && parent && parent.callee && parent.callee.property && fsWriteSymbols[parent.callee.property.name]) {
```

Comment:
Are you absolutely sure this code path ever gets hit? I would have thought it would have caught `typeof caleeValue.value === 'symbol'` given the use of `FS_FN`?

Note that an easy way to do the write detection in that case is just to have a `FS_WRITE_FN` symbol detected in the switch statement a few lines down.

Reply:
It hits this path, I added a test to cover this use case. It's function, I tested it on different cases and it was always function.